### DISCUSSION TITLE
Request retransmit: 4 / Nothing received, resend whole request

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#392] [Q&A]
Ist es normal, dass ich so viele retransmission errors bekome?

0:30:29.251 > Fetch inverter: xxxxxxxx
00:30:29.750 > Last missing
00:30:29.750 > Request retransmit: 4
00:30:29.880 > Last missing
00:30:29.880 > Request retransmit: 4
00:30:30.059 > Last missing
00:30:30.059 > Request retransmit: 4
00:30:30.197 > Last missing
00:30:30.197 > Request retransmit: 4
00:30:30.302 > Last missing
00:30:30.302 > Request retransmit: 4
00:30:30.470 > Success
00:30:31.221 > All missing
00:30:31.221 > Nothing received, resend whole request
00:30:32.008 > All missing
00:30:32.008 > Nothing received, resend whole request
00:30:32.828 > All missing
00:30:32.828 > Nothing received, resend whole request
00:30:33.550 > Last missing
00:30:33.550 > Request retransmit: 2
00:30:33.709 > Last missing
00:30:33.709 > Request retransmit: 2

**Comment by schlimmchen** [comment]
Da würde ich dir so gerne noch besser drauf antworten als "Ja und nein".
* Es ist natürlich ganz normal, dass Daten über eine Funkstrecke zeitweise nicht übertragen oder empfangen werden können. Die Luft ist ein geteiltes Medium, insb. im 2.4GHz Frequenzband kann es immer wieder zu Störungen kommen.
* Es ist nicht normal, dass das so häufig passiert. Auch bei mir passiert das immer noch deutlich zu häufig (für meinen Geschmack).
* Leider lässt sich das bisher nicht quantifizieren. Was fehlt sind Counter im OpenDTU bzw. in der Hoymiles Library, die zählen wie viele Antworten/ACKs nicht empfangen wurden, die erwartet wurden. Meine Meinung ist es, dass das ein großes Problem ist, dass die fehlen, denn so können wir schlecht beobachten, ob und wenn ja wann die "Verbindungsqualität" über die Zeit schwankt und insb. auch nicht festhalten, was denn der Optimalfall ist und auch nicht untereinander vergleichen.

Wie ist denn dein Setup? OpenDTU sehr nahe dran am Inverter, so wie bei mir unter 60cm? Ich hatte zuletzt Erfolg damit die Qualität spürbar zu verbessern, indem ich ein Dämpfungsglied in den Antennenpfad eingebaut habe.

![Screenshot from 2023-08-22 11-31-14](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3578416/6b411473-a5f8-4572-b5a0-58b310a46e64)

Mein OpenDTU-Fusion v2 Board mit der mitgelieferten Antenne konnte kaum Pakete mit dem Inverter austauschen. Ohne Antenne ging es dann. Das wird damit zu tun haben, dass OpenDTU und Inverter zu nah beieinander sind. Der Inverter "schreit" so laut, dass die OpenDTU nichts mehr "versteht". Wenn man die Antenne ganz abnimmt ist die Qualität schlechter als optimal, es gibt trotzdem Aussetzer, auch wenn es besser ist. Aber mit Dämpfungsglied wirds dann okay. Die Sendeleistung der OpenDTU lässt sich ja einstellen. Die muss man dann etwas erhöhen, weil das Dämpfungsglied nicht nur den Empfangspfad dämpft, sondern auch den Sendepfad -- geht ja über die gleiche Antenne.

Weiteres Indiz: Wenn das gleiche Board mit der mitgelieferten Antenne auf meinem Schreibtisch liegt, also ca. 7m weg mit zwei Holzwänden dazwischen, dann ist die Verbindung "perfekt": Praktisch keine Retransmits bzw. keine "* missing" Meldungen.

Danke für den Start dieser Diskussion, ich denke schon lange drüber nach so eine Diskussion zu starten und mich über dieses Thema auszutauschen, weil ich mit der Qualität der Datenübertragung leider nicht einverstanden bin. Ich bin gespannt, ob wir hier Fortschritte machen können.

**Comment by Manos1966** [comment]
Danke @schlimmchen fuer die ausfuehrliche Erklaerung! 👍 

Verdammt! die Open DTU ist 10cm vom HM-1500 entfernt [facepalm]
und ich habe das "grosse" Kommunikationsmodul mit externe Antenne, da hilf auch die Minimum (-18dBm) Sendeleistung wenig....

Einen Daempfer zu installieren ist eine sehr gute Idee, werde ich testen.

Vielleicht sollte ich vorab erwaehnen: Die Original DTU von Hoymiles arbeitet mit 15Min Sende Intervallen wenn ich mich richtig erinnere... :-)
 
Ich merke folgendes: Eine sehr hohe Anzahl von retransmits, staendig, sehr schnell und mit **sehr hohen Frequenz.**
Hier ein Beispiel (binnen drei Sekunden):
Uebrigens, sehe ich "etwas" wie eine laufende Nummer? "Request retransmit: 9" 
11:57:26.058 > Fetch inverter: xxxxxxxxxx
11:57:26.160 > TX RealTimeRunData Channel: 23 --> 15 74 40 20 87 80 16 00 72 80 0B 00 64 E4 86 86 00 00 00 00 00 00 00 00 E7 3F B1 
11:57:26.259 > Interrupt received
11:57:26.368 > RX Channel: 61 --> 95 74 40 20 87 74 40 20 87 01 00 01 01 E7 00 E3 00 E8 04 4E 04 6A 00 01 8C D8 09 | -80 dBm
11:57:26.578 > RX Period End
11:57:26.578 > Last missing
11:57:26.578 > Request retransmit: 2
11:57:26.578 > TX RequestFrame Channel: 40 --> 15 74 40 20 87 80 16 00 72 82 E0 
11:57:26.691 > Interrupt received
11:57:26.776 > RX Channel: 75 --> 95 74 40 20 87 74 40 20 87 81 00 01 70 C0 A5 | -80 dBm
11:57:26.875 > RX Period End
11:57:26.875 > ERROR in RealTimeRunData: Received fragment size: 4, min expected size: 62
11:57:26.875 > Packet handling error
11:57:27.078 > TX AlarmData Channel: 61 --> 15 74 40 20 87 80 16 00 72 80 11 00 64 E4 86 86 00 00 00 00 00 00 00 00 3D 24 6A 
11:57:27.496 > RX Period End
11:57:27.496 > All missing
11:57:27.496 > Nothing received, resend whole request
11:57:27.496 > TX AlarmData Channel: 75 --> 15 74 40 20 87 80 16 00 72 80 11 00 64 E4 86 86 00 00 00 00 00 00 00 00 3D 24 6A 
11:57:27.593 > Interrupt received
11:57:27.695 > RX Channel: 40 --> 95 74 40 20 87 74 40 20 87 01 00 01 B0 01 00 01 93 04 93 04 00 00 00 00 80 02 A7 | -80 dBm
11:57:27.896 > Interrupt received
11:57:28.001 > RX Channel: 23 --> 95 74 40 20 87 74 40 20 87 07 F2 15 80 02 19 A9 8C 00 8C 00 00 00 0D E8 80 02 20 | -80 dBm
11:57:28.207 > Interrupt received
11:57:28.313 > RX Channel: 75 --> 95 74 40 20 87 74 40 20 87 08 19 AA 8C 01 8C 01 FF FF F2 18 80 02 19 AB 8C 01 79 | -80 dBm
11:57:28.507 > RX Period End
11:57:28.507 > Last missing
11:57:28.507 > Request retransmit: 9
11:57:28.507 > TX RequestFrame Channel: 3 --> 15 74 40 20 87 80 16 00 72 89 EB 
11:57:28.558 > Interrupt received
11:57:28.618 > RX Channel: 40 --> 95 74 40 20 87 74 40 20 87 09 8C 01 00 00 0D E8 80 02 19 AC 8C 02 8C 02 FF FF C3 | -80 dBm
11:57:28.710 > RX Period End
11:57:28.710 > Last missing
11:57:28.710 > Request retransmit: 10
11:57:28.710 > TX RequestFrame Channel: 23 --> 15 74 40 20 87 80 16 00 72 8A E8 
11:57:28.759 > Interrupt received
11:57:28.817 > RX Channel: 61 --> 95 74 40 20 87 74 40 20 87 0A F2 19 80 02 19 AD 8C 03 8C 03 00 00 0D E6 80 02 2B | -80 dBm
11:57:29.035 > RX Period End
11:57:29.035 > Last missing
11:57:29.035 > Request retransmit: 11
11:57:29.035 > TX RequestFrame Channel: 40 --> 15 74 40 20 87 80 16 00 72 8B E9 
11:57:29.127 > RX Period End
11:57:29.127 > Last missing
11:57:29.127 > Request retransmit: 11
11:57:29.127 > TX RequestFrame Channel: 61 --> 15 74 40 20 87 80 16 00 72 8B E9 
11:57:29.231 > RX Period End
11:57:29.231 > Last missing
11:57:29.231 > Request retransmit: 11
11:57:29.231 > TX RequestFrame Channel: 75 --> 15 74 40 20 87 80 16 00 72 8B E9 
11:57:29.331 > RX Period End
11:57:29.331 > Last missing
11:57:29.331 > Retransmit timeout

**Reply by swingstate** [reply]
Gleiches Thema hier. DTU & HM sind 70cm von einander entfernt und es gibt nach meinem Geschmack zu viele Retransmit's. So einen Dämpfer werde ich mal testen.

**Comment by schlimmchen** [comment]
Ich möchte ein Update geben: Inzwischen fahre ich mit 30+6dB Dämpfer und der Sendeleistung "Minimum" sowie einer der einfachen Stabantennen sehr gut. Neu hinzu kommt allerdings, dass ab und zu ein "DTU Command failed" auftaucht im Log. Wahrscheinlich ist die Dämpfung jetzt etwas zu stark oder die Sendeleistung etwas zu gering?! Ich weiß, dass sich die Qualität wieder verschlechtert (mehr retransmits) wenn ich die Sendeleistung erhöhe.

Ich probiere als nächstes minimale Sendeleitung und 30dB Dämpfung. Diese Kombination hatte ich noch nicht.

**Reply by Manos1966** [reply]
Ich habe den BECEN Golden 2 W 6 GHz 50 Ohm 30dB Stecker(30dB) geholt, ich sehe (oder glaube zu sehen) leider keinen Unterschied...

**Reply by Manos1966** [reply]
Ich habe jetzt das grosse Kommunikationsmodul mit Antenne, durch ein kleines (ohne Antenne) ausgetauscht.
Jetzt habe ich deutlich weniger re-transmits) 👍 
![20230826_104833 - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/718fcbdf-4dd2-42fb-b107-a5ebfbf2566c)

**Reply by schlimmchen** [reply]
> ich sehe (oder glaube zu sehen) leider keinen Unterschied...

Schade...

Was sehen wir denn auf dem Foto? Das alte große Modul oder das neue kleine? Und betreibst du es mit Dämpfung oder ohne?

Jedenfalls schön, dass du Fortschritte machst.

**Reply by Manos1966** [reply]
Rechts das alte grosse Modul, links das neue.
Ich habe das alte Grosse Modul mit einem 30dB Daempfer probiert, sah ich aber keine Verbesserung.
Jetzt mit dem kleinen Modul habe ich eine deutliche Verbesserung.
![20230828_011807 - Copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/6ba5e885-46e9-49f4-aa2f-198fbbd035ee)

**Comment by dk7td** [comment]
Ganz ohne Antenne ist eine sehr schlechte Idee, auch wenn’s nur kurzzeitig ist. Dabei kann das Sendemodul schneller sterben, als einem lieb ist. Es bekommt seine Sendeleistung nicht los, diese reflektiert von der offenen Antennenbuchse zurück ins Sendemodul und das war’s dann u.U.
